lib/matplotlib/widgets.py

```diff
@@ -2115,9 +2115,10 @@ def __init__(self, ax, onselect, drawtype='box',
         self.maxdist = maxdist
 
         if rectprops is None:
-            props = dict(mec='r')
+            props = dict(markeredgecolor='r')
         else:
-            props = dict(mec=rectprops.get('edgecolor', 'r'))
+            props = dict(markeredgecolor=rectprops.get('edgecolor', 'r'))
+        props.update(marker_props if marker_props is not None else {})
         self._corner_order = ['NW', 'NE', 'SE', 'SW']
         xc, yc = self.corners
         self._corner_handles = ToolHandles(self.ax, xc, yc, marker_props=props,
```

Comment:
Wouldn't it be better to pass `marker_props` here?  I think that is what @harupy  is suggesting at https://github.com/matplotlib/matplotlib/issues/12027#issuecomment-471242158